ed25519/src/lib.rs

```diff
@@ -140,29 +110,30 @@ impl Pair {
 		Public(r)
 	}
 }
-impl Signature {
-	/// Verify a message.
-	pub fn verify(&self, message: &[u8], public: &Public) -> bool {
-		let peer_public_key = untrusted::Input::from(&public.0[..]);
-		let msg = untrusted::Input::from(message);
-		let sig = untrusted::Input::from(&self.0[..]);
-
-		match signature::verify(&signature::ED25519, peer_public_key, msg, sig) {
-			Ok(_) => true,
-			_ => false,
-		}
+
+/// Verify a signature on a message.
+pub fn verify_strong(sig: &Signature, message: &[u8], pubkey: &Public) -> bool {
+	let public_key = untrusted::Input::from(&pubkey.0[..]);
+	let msg = untrusted::Input::from(message);
+	let sig = untrusted::Input::from(&sig.0[..]);
+
+	match signature::verify(&signature::ED25519, public_key, msg, sig) {
+		Ok(_) => true,
+		_ => false,
 	}
 }
+
 impl From<&'static str> for Public {
 	fn from(hex: &'static str) -> Self {
 		let mut r = [0u8; 32];
-		r.copy_from_slice(&FromHex::from_hex(hex).unwrap()[0..32]);
+		let v: Vec<_> = FromHex::from_hex(hex).unwrap();
+		r.copy_from_slice(&v[0..32]);
 		Public(r)
 	}
 }
 impl From<&'static str> for Pair {
 	fn from(hex: &'static str) -> Self {
-		let data = FromHex::from_hex(hex).expect("Key pair given is static so hex should be good.");
+		let data: Vec<_> = FromHex::from_hex(hex).expect("Key pair given is static so hex should be good.");
```

Comment:
would have been nice to replace this code with `hex![...]` macro

Reply:
don't think that' possible; there is a difference between a `&'static str` and a string literal.